encodings/alp/src/alp/array.rs

```diff
@@ -46,6 +46,14 @@ impl ALPArray {
         let mut children = Vec::with_capacity(2);
         children.push(encoded);
         if let Some(patches) = &patches {
+            if patches.dtype() != &dtype {
+                vortex_bail!(MismatchedTypes: dtype, patches.dtype());
+            }
+
+            if patches.values().validity_mask()?.false_count() != 0 {
```

Comment:
Calling validity_mask here triggers a "canonicalization" of the validity buffer. You should instead use `patches.values().all_valid()?` which should short-circuit if possible

Reply:
Done, thanks.